### ISSUE TITLE
Email Validation / Rules

### ISSUE THREAD

**Issue by atkirtland** [learning-unlimited/ESP-Website#3639]
It would be helpful if we could require the emails used for sign ups to follow certain rules, for example requiring them to be `*@(school name).edu` addresses.

**Comment by willgearty** [comment]
Thanks for the issue @atkirtland. I'm guessing you mean for "teacher" accounts? As opposed to "student" or "guardian" accounts?

**Comment by atkirtland** [comment]
That's right. I think "teacher" accounts would be the most helpful use case.

**Comment by willgearty** [comment]
As someone who often uses their gmail account rather than their institution account, I'm not sure this is a great idea, even if only for teacher accounts. I can certainly see the potential benefits (e.g., requiring a university email address ensures that teachers are actually associated with the university), but I can see lots of reasons teachers may want to use other email addresses. However, I suppose if this is an optional tag, it would then be up to the chapter to weigh these pros/cons and decide themselves.

**Comment by SuyashJain17** [comment]
I’ve been thinking about this, and maybe there’s a middle ground that keeps it flexible.

Instead of hard-requiring `.edu` addresses globally, we could make email restrictions optional and configurable per chapter, and only apply them to teacher accounts.

For example, a chapter could enable a setting like “enforce teacher email rules,” and then define:

A list of allowed domains (e.g. `school.edu`)

Or optionally a regex pattern for more advanced cases

And possibly choose between “block” mode (hard requirement) or “warn only” mode

That way, chapters that want stricter institutional verification can enable it, and others can leave it off. Existing accounts wouldn’t be affected — the rules would only apply to new signups or when a teacher updates their email.

From an implementation standpoint, this could live in chapter-level configuration, with validation added to the teacher registration and profile update forms.

If this sounds reasonable, I’m happy to sketch out a concrete implementation aligned with current ESP patterns.

@willgearty @atkirtland

**Comment by willgearty** [comment]
Yeah, this sounds like a good plan. I think have a configurable setting with a list of domains or regex would be great. There's also probably some overlap here with #3915
